### ISSUE TITLE
 Could not find artifacts for Solidity tests (Windows only, v4.1.5)

### ISSUE THREAD

**Issue by woodenchalet** [trufflesuite/truffle#885]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue
  In Version v4.1.5 truffle test will throw errors, If I downgrade the truffle version to v4.0.4 , it will work

  TestMyContract
    1) "before all" hook: prepare suite
  0 passing (625ms)
  1 failing

  1) TestMyContract "before all" hook: prepare suite:
     Error: Could not find artifacts for "path"

## Steps to Reproduce

Truffle test 

## Environment

* Operating System: Windows 10
* Truffle version (`truffle version`): v4.1.5

**Comment by ismaelbej** [comment]
For me the same fails with 4.1.5 but works ok with v4.1.4 (Windows 7 x64).

**Comment by cgewecke** [comment]
@ismaelbej Are you seeing exactly the same error?

**Comment by ismaelbej** [comment]
I'll try to create a small reproducible example for now the stack trace output

```
   1) TestRelay "before all" hook: prepare suite:
      Error: Could not find artifacts for /D/test/contracts/Parser/Tx.sol from any sources
       at Resolver.require (C:\Users\little\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-resolver\index.js:37:1)
       at TestResolver.require (C:\Users\little\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\testresolver.js:17:1)
       at TestResolver.require (C:\Users\little\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\testresolver.js:17:1)
       at C:\Users\little\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\soliditytest.js:141:1
       at Array.forEach (<anonymous>)
       at C:\Users\little\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\soliditytest.js:140:1
       at C:\Users\little\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-deployer\src\deferredchain.js:20:1
       at <anonymous>
       at process._tickCallback (internal/process/next_tick.js:188:7)
 
```

The file `Parser/Tx.sol`  is a dependency of `RelayTests` but compilation and migration run fine. Also executing javascript test will run without issues.

And the solidity test, but I think the content doesn't matter too much because it is never run.

```
pragma solidity ^0.4.15;

import "truffle/Assert.sol";
import "truffle/DeployedAddresses.sol";
import "../contracts/RelayTests.sol";

contract TestRelay {

    function testFlip32Bytes() {
        RelayTests dr = RelayTests(DeployedAddresses.RelayTests());
        uint expected = 0x201f1e1d1c1b1a191817161514131211100f0e0d0c0b0a090807060504030201;
        Assert.equal(dr.flip32BytesPublic(0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20), expected, "flip32Bytes failed");
    }

}

```

**Comment by cgewecke** [comment]
@ismaelbej Awesome, thank you. That's perfect.

**Comment by woodenchalet** [comment]
@cgewecke Could be a PowerShell or Windows thing, I try to run the truffle test under Linux, they all passed. 

![image](https://user-images.githubusercontent.com/7042874/38440704-43c947be-39b0-11e8-9601-2fdc449883ac.png)

![image](https://user-images.githubusercontent.com/7042874/38440721-4fb93f7a-39b0-11e8-97b8-ad5d4913ace7.png)

**Comment by cgewecke** [comment]
@woodenchalet Ok great, good to know. I'm going to modify the title of your issue to reflect that and look into why the paths are being made OS neutral correctly anymore.

**Comment by franva** [comment]
Hi @cgewecke I am facing the exactly same issue here.

Could you please update what's happening ?

Hi @woodenchalet , @ismaelbej 
Is there any workaround?

Thanks

**Comment by franva** [comment]
I downgraded to truffle 4.1.14 and it works.

Btw, this is on Windows platform.

**Comment by ismaelbej** [comment]
After some debugging `compileContractsWithTestFilesIfNeeded` in https://github.com/trufflesuite/truffle-core/blob/develop/lib/test.js#L103 will return dependencies in different format between v4.1.4 and v4.1.5 on windows

With v4.1.5
```
{
  "paths": [
    "/C/Users/daffyduck/develop/mud/contracts/ConvertLib.sol",
    "/C/Users/daffyduck/develop/mud/contracts/MetaCoin.sol",
    "/C/Users/daffyduck/develop/mud/test/TestMetacoin.sol",
    "truffle/Assert.sol",
    "truffle/DeployedAddresses.sol"
  ]
}
```

With v4.1.4
```
{
  "paths": [
    "C:\\Users\\daffyduck\\develop\\mud\\test\\TestMetacoin.sol",
    "truffle/Assert.sol",
    "truffle/DeployedAddresses.sol",
    "C:\\Users\\daffyduck\\develop\\mud\\contracts\\MetaCoin.sol",
    "C:\\Users\\daffyduck\\develop\\mud\\contracts\\ConvertLib.sol"
  ]
}
```

**Comment by cgewecke** [comment]
@ismaelbej Thanks! That's really helpful. We're going to spin up our windows box and get to the bottom this. . .hopefully fix this quickly.

**Comment by jrmillerwork** [comment]
I am also experiencing this problem with Windows using Powershell, even after I change to pragma solidity ^0.4.14; One difference I see is what is missing is ConvertLib (which seems to be different from what is mentioned above):
Error: Could not find artifacts for /C/Users/jeff/smartContracts/MetaCoin/contracts/ConvertLib.sol from any sources
      at Resolver.require (C:\Users\jeff\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-resolver\index.js:37:1)
      at TestResolver.require (C:\Users\jeff\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\testresolver.js:17:1)
      at TestResolver.require (C:\Users\jeff\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\testresolver.js:17:1)
      at C:\Users\jeff\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\soliditytest.js:141:1
      at Array.forEach (<anonymous>)
      at C:\Users\jeff\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-core\lib\testing\soliditytest.js:140:1
      at C:\Users\jeff\AppData\Roaming\npm\node_modules\truffle\build\webpack:\~\truffle-deployer\src\deferredchain.js:20:1
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)

I installed Truffle within the past week. Finally, to be clear, the js test works; it is only the Solidity test that fails.

**Comment by gnidan** [comment]
Oops, my bad for breaking this. ^ PR should fix, expect release early next week. Thanks!

**Comment by gnidan** [comment]
This has been released in v4.1.6. Thanks!
